solang-parser/src/helpers/fmt.rs

```diff
@@ -0,0 +1,2518 @@
+        f.write_char(' ')?;
+        self.execution_block.fmt(f)
+    }
+}
+
+impl Display for pt::YulFunctionCall {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result {
+        self.id.fmt(f)?;
+        f.write_char('(')?;
+        write_separated(&self.arguments, f, ", ")?;
+        f.write_char(')')
+    }
+}
+
+impl Display for pt::YulFunctionDefinition {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result {
+        f.write_str("function ")?;
+        self.id.fmt(f)?;
+        f.write_char('(')?;
+        write_separated(&self.params, f, ", ")?;
+        f.write_str(") ")?;
+
+        if !self.returns.is_empty() {
+            f.write_str("-> (")?;
+            write_separated(&self.returns, f, ", ")?;
+            f.write_str(") ")?;
+        }
+
+        self.body.fmt(f)
+    }
+}
+
+impl Display for pt::YulSwitch {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result {
+        f.write_str("switch ")?;
+        self.condition.fmt(f)?;
+        if !self.cases.is_empty() {
+            f.write_char(' ')?;
+            write_separated(&self.cases, f, " ")?;
+        }
+        write_opt!(f, " ", &self.default);
+        Ok(())
+    }
+}
+
+impl Display for pt::YulTypedIdentifier {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result {
+        self.id.fmt(f)?;
+        write_opt!(f, ": ", &self.ty);
+        Ok(())
+    }
+}
+
+// enums
+impl Display for pt::CatchClause {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result {
+        match self {
+            Self::Simple(_, param, block) => {
+                f.write_str("catch ")?;
+                write_opt!(f, '(', param, ") ");
+                block.fmt(f)
+            }
+            Self::Named(_, ident, param, block) => {
+                f.write_str("catch ")?;
+                ident.fmt(f)?;
+                f.write_char('(')?;
+                param.fmt(f)?;
+                f.write_str(") ")?;
+                block.fmt(f)
+            }
+        }
+    }
+}
+
+impl Display for pt::Comment {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result {
+        f.write_str(self.value())
+    }
+}
+
+impl Display for pt::ContractPart {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result {
+        match self {
+            Self::StructDefinition(inner) => inner.fmt(f),
+            Self::EventDefinition(inner) => inner.fmt(f),
+            Self::EnumDefinition(inner) => inner.fmt(f),
+            Self::ErrorDefinition(inner) => inner.fmt(f),
+            Self::VariableDefinition(inner) => inner.fmt(f),
+            Self::FunctionDefinition(inner) => inner.fmt(f),
+            Self::TypeDefinition(inner) => inner.fmt(f),
+            Self::Annotation(inner) => inner.fmt(f),
+            Self::Using(inner) => inner.fmt(f),
+            Self::StraySemicolon(_) => f.write_char(';'),
+        }
+    }
+}
+
+impl Display for pt::ContractTy {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result {
+        f.write_str(self.as_str())
+    }
+}
+impl pt::ContractTy {
```

Comment:
```suggestion
}

impl pt::ContractTy {
```

Reply:
Display `impl` is grouped with its corresponding struct `impl`, but I can change that if you want to

Reply:
Well I thought this slipped by. Anyways it's a nitpick, I'll leave it up to you